geo_autoRIFT/autoRIFT/autoRIFT.py

```diff
@@ -29,61 +29,98 @@
 # Note: this is based on the MATLAB code, "auto-RIFT", written by Alex S. Gardner,
 #       and has been translated to Python and further optimized.
 #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+import random
 
-
-
-import pdb
-import subprocess
-import re
-import string
+import cv2
 import sys
-
+import numpy as np
+from scipy.ndimage.morphology import distance_transform_edt
+from scipy.ndimage.filters import generic_filter
 
 
 class autoRIFT:
-    '''
+    """
     Class for mapping regular geographic grid on radar imagery.
-    '''
-    
-    
-    
+    """
+    def preprocess_filt_wal_nodata_fill(self):
+        """
+        Wallis filter with nodata infill for L7 SLC Off preprocessing
+        """
+        self.zeroMask = np.full(self.I1.shape, True)
+        for image in (self.I1, self.I2):
+            zero_mask_1 = image == 0
+            buff = np.sqrt(2 * ((self.WallisFilterWidth - 1) / 2) ** 2) + 0.01
+
+            # find edges of image, this makes missing scan lines valid and will
+            # later be filled with random white noise
+            missing_data = distance_transform_edt(zero_mask_1) < 30
+            missing_data = missing_data & zero_mask_1
+            missing_data = distance_transform_edt(~missing_data) <= buff
+
+            # trying to frame out the image
+            valid_domain = ~zero_mask_1 | missing_data
+            self.zeroMask &= ~valid_domain
+
+            low_std = generic_filter(image, np.std, size=self.WallisFilterWidth) < self.StandardDeviationCutoff
```

Comment:
When profiling preprocess_filt_wal_nodata_fill, this call was found to take 98% of the computation time.